pkg/resourceutil/resource.go

```diff
@@ -82,7 +83,10 @@ func NewResources(filepath, namespace string) ([]Resource, error) {
 		return nil, err
 	}
 
-	for _, resourceYAML := range strings.Split(string(stream), "---") {
+	// split resources on --- yaml document delimiter
+	re := regexp.MustCompile(`\n---\n`)
```

Comment:
In this way, if data would be:

```
data:
  key1: aaaa---dldl
  a.json: "{"key":"\n---\nfoo\nbar\n---\n"}"
```
?

It should be better to use another regex to take only start of line? So something like `^---\n` Or `^---$`?

Reply:
I started with that idea too, but it does not work because we have a single string containing all the file. So here `^` represents the start of file, and `$` the end.  
Searching for \n before and after `---` makes sure that is a single line containing only `---`

Reply:
the example does not match this regex because \n in that example is literal \ and n, and not the EOL